servers/republik/modules/crowdfundings/lib/scheduler/changeover.js

```diff
@@ -61,31 +69,53 @@ const changeover = async (args, { pgdb, mail: { enforceSubscriptions } }) => {
       }
 
       const activeMembership = userMemberships.find(m => m.active)
+      if (moment(getLastEndDate(activeMembership.periods)) > endDate) {
+        console.error(
+          'changeover failed: active memberships ends later',
+          { user, endDate }
+        )
+        return
+      }
+
       const dormantMemberships = findDormantMemberships({
         memberships: userMemberships,
         user: { id: user.id }
       })
 
-      // End here if there is not dormant membership to be found
+      // End here if there is no dormant membership to be found
       if (dormantMemberships.length < 1) {
         return
       }
 
-      // Elect a dormant membership to activate. Rule is to elect dorman
-      // membership with lowest sequenceNumber
-      const electedDormantMembership = dormantMemberships.reduce(
-        (acc, curr) =>
-          !acc || curr.sequenceNumber < acc.sequenceNumber
-            ? curr
-            : acc
-      )
+      // Elect a dormant membership to activate. Rule is to elect dormant
+      // membership with lowest sequenceNumber.
+      // Sorts sequenceNumber ascending, uses first row. Will overwrite, if a
+      // membershipType BENEFACTOR_ABO comes by.
+      const electedDormantMembership = dormantMemberships
+        .sort((a, b) => ascending(a.sequenceNumber, b.sequenceNumber))
+        .reduce((acc, curr) =>
+          (!acc && curr) ||
+          (acc.membershipType.name !== 'BENEFACTOR_ABO' && curr.membershipType.name === 'BENEFACTOR_ABO' && curr) ||
+          acc
+        )
 
-      const transaction = await pgdb.transactionBegin()
       debug({
         activeMembership: activeMembership.id,
-        electedDormantMembership: electedDormantMembership.id
+        electedDormantMembership: {
+          id: electedDormantMembership.id,
+          type: electedDormantMembership.membershipType.name
+        },
+        runDry
       })
 
+      stats.changeover++
+
+      if (runDry) {
+        return
```

Comment:
please `console.log` so that testing users know what's going on.

Reply:
Printing [a dry-run warning](https://github.com/orbiting/backends/pull/176/commits/4f58cca8a97c444cab67e87d6205fc6b8495af95)